samples/javascript/powerapps-pcf/PowerAppLinerInput/LinearInputControl/index.ts

```diff
@@ -0,0 +1,179 @@
+    private inputElement: HTMLInputElement;
+    private _container: HTMLDivElement;
+    private _context: ComponentFramework.Context<IInputs>;
+    private _refreshData: EventListenerOrEventListenerObject;
+    private _ws: WebSocket;
+    private _switcher: boolean;
+    /**
+     * Empty constructor.
+    */
+    constructor() {}
+
+    /**
+     * Used to initialize the control instance. Controls can kick off remote server calls 
+         and other initialization actions here.
+    * Data-set values are not initialized here, use updateView.
+    * @param context The entire property bag available to control via Context Object; 
+        It contains values as set up by the customizer mapped to property names defined 
+        in the manifest, as well as utility functions.
+    * @param notifyOutputChanged A callback method to alert the framework that the 
+        control has new outputs ready to be retrieved asynchronously.
+    * @param state A piece of data that persists in one session for a single user. 
+        Can be set at any point in a controls life cycle by calling 'setControlState' 
+        in the Mode interface.
+    * @param container If a control is marked control-type='standard', it will receive 
+        an empty div element within which it can render its content.
+    */
+    public async init(
+        context: ComponentFramework.Context<IInputs>,
+        notifyOutputChanged: () => void,
+        state: ComponentFramework.Dictionary,
+        container: HTMLDivElement
+    ): Promise<void> {
+        // single place to switch wps.
+        this._switcher = true;
+
+        // Add control initialization code
+        this._context = context;
+        this._container = document.createElement("div");
+        this._notifyOutputChanged = notifyOutputChanged;
+        this._refreshData = this.refreshData.bind(this);
+
+        // creating HTML elements for the input type range and binding it to the function which 
+        // refreshes the control data
+        this.inputElement = document.createElement("input");
+        this.inputElement.setAttribute("type", "range");
+        this.inputElement.addEventListener("input", this._refreshData);
+
+        //setting the max and min values for the control.
+        this.inputElement.setAttribute("min", "1");
+        this.inputElement.setAttribute("max", "1000");
+        this.inputElement.setAttribute("class", "linearslider");
+        this.inputElement.setAttribute("id", "linearrangeinput");
+
+        // creating a HTML label element that shows the value that is set on the linear range control
+        this.labelElement = document.createElement("label");
+        this.labelElement.setAttribute("class", "LinearRangeLabel");
+        this.labelElement.setAttribute("id", "lrclabel");
+
+        if (this._switcher)
+        {
+            // set up websocket connection to WPS service.
+            const response = await fetch("http://localhost:7071/api/negotiate");
```

Comment:
is relative url working? Otherwise we might need a setting to configure the URL?

Reply:
It's a simple local sample. When deploy to cloud, they're hosted under different platform, Azure Functions and Power Apps. So not able to under same host anyway and it's up to customer how to setup this connection. The best solution is not decided yet.